R/derive_extreme_event.R

```diff
@@ -486,6 +482,248 @@ derive_extreme_event <- function(dataset,
+#'   "PILOT01",    "LB", "01-1211",      8, "2012-10-30T14:26",
+#'   "PILOT01",    "LB", "01-1211",    162, "2013-01-08T12:13",
+#'   "PILOT01",    "LB", "09-1081",     47, "2014-02-01T10:55",
+#'   "PILOT01",    "LB", "09-1081",    219, "2014-05-10T11:15",
+#'   "PILOT01",    "LB", "09-1088",    283, "2014-09-27T12:13",
+#'   "PILOT01",    "LB", "09-1088",    322, "2014-10-09T13:25"
+#' ) %>%
+#'   mutate(
+#'     ADT = convert_dtc_to_dt(LBDTC)
+#'   )
+#'
+#' derive_vars_extreme_event(
+#'   adsl,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   events = list(
+#'     event(
+#'       dataset_name = "adsl",
+#'       condition = !is.na(DTHDT),
+#'       set_values_to = exprs(LSTALVDT = DTHDT, DTHFL = "Y")
+#'     ),
+#'     event(
+#'       dataset_name = "lb",
+#'       condition = !is.na(ADT),
+#'       order = exprs(ADT),
+#'       mode = "last",
+#'       set_values_to = exprs(LSTALVDT = ADT, DTHFL = "N")
+#'     ),
+#'     event(
+#'       dataset_name = "adsl",
+#'       condition = !is.na(TRTEDT),
+#'       order = exprs(TRTEDT),
+#'       mode = "last",
+#'       set_values_to = exprs(LSTALVDT = TRTEDT, DTHFL = "N")
+#'     )
+#'   ),
+#'   source_datasets = list(adsl = adsl, lb = lb),
+#'   order = exprs(USUBJID, LSTALVDT, event_nr),
+#'   mode = "last",
+#'   keep_source_vars = exprs(STUDYID, USUBJID),
+#'   new_vars = exprs(LSTALVDT = LSTALVDT, DTHFL = DTHFL)
+#' )
+#'
+#' # Derive DTHCAUS from AE and DS domain data
+#' adsl <- tribble(
+#'   ~STUDYID,  ~USUBJID,
+#'   "STUDY01", "PAT01",
+#'   "STUDY01", "PAT02",
+#'   "STUDY01", "PAT03"
+#' )
+#' ae <- tribble(
+#'   ~STUDYID,  ~USUBJID, ~AESEQ, ~AEDECOD,       ~AEOUT,  ~AEDTHDTC,
+#'   "STUDY01", "PAT01",  12,     "SUDDEN DEATH", "FATAL", "2021-04-04",
+#'   "STUDY01", "PAT01",  13,     "CARDIAC ARREST", "FATAL", "2021-04-03",
+#' )
+#'
+#' ds <- tribble(
+#'   ~STUDYID, ~USUBJID, ~DSSEQ, ~DSDECOD, ~DSTERM, ~DSSTDTC,
+#'   "STUDY01", "PAT02", 1, "INFORMED CONSENT OBTAINED", "INFORMED CONSENT OBTAINED", "2021-04-03",
+#'   "STUDY01", "PAT02", 2, "RANDOMIZATION", "RANDOMIZATION", "2021-04-11",
+#'   "STUDY01", "PAT02", 3, "DEATH", "DEATH DUE TO PROGRESSION OF DISEASE", "2022-02-01",
+#'   "STUDY01", "PAT03", 1, "DEATH", "POST STUDY REPORTING OF DEATH", "2022-03-03"
+#' )
+#'
+#' derive_vars_extreme_event(
+#'   adsl,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   events = list(
+#'     event(
+#'       dataset_name = "ae",
+#'       condition = AEOUT == "FATAL",
+#'       set_values_to = exprs(DTHCAUS = AEDECOD, DTHDT = convert_dtc_to_dt(AEDTHDTC)),
+#'       order = exprs(DTHDT)
+#'     ),
+#'     event(
+#'       dataset_name = "ds",
+#'       condition = DSDECOD == "DEATH" & grepl("DEATH DUE TO", DSTERM),
+#'       set_values_to = exprs(DTHCAUS = DSTERM, DTHDT = convert_dtc_to_dt(DSSTDTC)),
+#'       order = exprs(DTHDT)
+#'     )
+#'   ),
+#'   source_datasets = list(ae = ae, ds = ds),
+#'   tmp_event_nr_var = event_nr,
+#'   order = exprs(USUBJID, DTHDT, event_nr),
```

Comment:
No need to specify `USUBJID` because it is already in `by_vars`.

Reply:
Updated